md/sv2-workshop.md

```diff
@@ -16,38 +16,59 @@ http://75.119.150.111:8888/html/sv2-workshop.html
 ---
 
 ## Prerequisites
+The required programs are:
+1. `bitcoin-core` fork with Sv2 support.
+2. `stratum` repo with roles logic.
+3. `cpuminer` to act as a hasher (miner) (Miner Role only).
 
-### Install Rust:
-```
-curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
-```
+There are two ways to get setup:
+1. With a Docker image that contains all the required packages and programs (recommended).
+2. By manually installing the required packages and programs.
 
-### Clone & Build SRI
+---
 
-```
-cd $HOME
-git clone https://github.com/stratum-mining/stratum
-cd stratum
-git checkout workshop
-```
+## Method 1: Docker (Recommended)
+1. Install [Docker](https://docs.docker.com/engine/install/).
+2. Configure Docker with the following minimum resource allocations:
+    - CPU limit: 10
+    - Memory limit: 8GB
+    - Swap: 1GB
+    - Virtual disk limit: 64 GB
+3. Run the image: `docker run -it --rm sv2-workshop:latest`
```

Comment:
```
$ docker run -it --rm sv2-workshop:latest
Unable to find image 'sv2-workshop:latest' locally
docker: Error response from daemon: pull access denied for sv2-workshop, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
```

Reply:
Try now with these commands and lmk how it goes:

```sh
docker pull rrybarczyk/sv2-workshop:latest
docker run -it --rm rrybarczyk/sv2-workshop:latest
```

Reply:
works!

Reply:
only on MacOS M2 tough

I tried on a x86-64 Ubuntu and got this:
```
$ docker pull rrybarczyk/sv2-workshop:latest
latest: Pulling from rrybarczyk/sv2-workshop
4ce000a43472: Pull complete 
4e4b36411342: Pull complete 
c26a44c50f9e: Pull complete 
cd91f28aa699: Pull complete 
59164fdac9b1: Pull complete 
71ea2e4f1c41: Pull complete 
5a8b82b61c55: Pull complete 
219508e87ada: Pull complete 
a2df3a6866d9: Pull complete 
cea5451fedd2: Pull complete 
Digest: sha256:4e1775ebb68ae620800d145ce67ab82f4b0efff09e2a57f03e591f1c47448f25
Status: Downloaded newer image for rrybarczyk/sv2-workshop:latest
docker.io/rrybarczyk/sv2-workshop:latest
$ docker run -it --rm rrybarczyk/sv2-workshop:latest
WARNING: The requested image's platform (linux/arm64/v8) does not match the detected host platform (linux/amd64/v3) and no specific platform was requested
exec /usr/bin/bash: exec format error
```

I get that the original goal was to bypass the hiccups on MacOS arm, but I wonder if it wouldn't be beneficial to extend the Docker setup to x86-64 users as well?

Reply:
That is a great idea. 

I figured out how to make the `sv2-workshop` image for x86-64 based linux machines and rebuilt it and pushed to docker hub.

Can you try it out again and lmk if that fixed it?

```sh
docker pull rrybarczyk/sv2-workshop:latest
docker run -it --rm rrybarczyk/sv2-workshop:latest
```

Reply:
it works!